onnxruntime/test/ir/graph_test.cc

```diff
@@ -700,6 +692,125 @@ TEST_F(GraphTest, GraphConstruction_CheckInputNodeOrderMaintained) {
   }
 }
 
+TEST_F(GraphTest, GraphConstruction_PriorityBasedTopologicalSort_CompressDecompress) {
+  Model model("graph_1", false, *logger_);
+  auto& graph = model.MainGraph();
+
+  /*
+                          |
+                  node_0 (Identity)
+                      /      \
+        node_1 (Identity)   compress (pri = -10)
+                    |         |
+        node_4 (Identity)  decompress (pri = 10)
+                      \       /
+                      node_5 (Merge)
+                          |                   
+  */
```

Comment:
I like these new tests. Could we enhance them a bit to simulate multiple layers? I'd like to make sure the orders in  multiple-layer are what we expected.

Reply:
done.